### DISCUSSION TITLE
使用 Vue 节点时，如何在创建节点时向组件传入 props?

### DISCUSSION THREAD

**Discussion by Lzz1027** [antvis/X6#4150] [Q&A]
在官网文档中没有找到明确提示，antv/x6-vue-shape 的 register 方法中好像也没有类似 React 组件中 effect 字段的功能

**Comment by TLovers** [comment]
文档上有写

**Comment by lilinhao1054** [comment]
这样应该能有类似的效果
```js
const count = ref(0)

const getNode: any = inject('getNode')

onMounted(() => {
  const node = getNode() as Node
  const data = node.getData()
  count.value = data.count
  node.on('change:data', ({ current }) => {
    count.value = current.count
  })
})
```

**Comment by zlecheng** [comment]
> 在官网文档中没有找到明确提示，antv/x6-vue-shape 的 register 方法中好像也没有类似 React 组件中 effect 字段的功能

可以借鉴下这篇文档里面的方法：https://www.yuque.com/sxd_panda/antv/vue-node

![image](https://github.com/user-attachments/assets/51b1e78a-00eb-41c2-ad77-e750bfeb80df)
